ipn/ipnlocal/serve.go

```diff
@@ -469,6 +466,37 @@ func (b *LocalBackend) proxyHandlerForBackend(backend string) (*httputil.Reverse
 	return rp, nil
 }
 
+func addProxyForwardedHeaders(r *httputil.ProxyRequest) {
+	r.Out.Header.Set("X-Forwarded-Host", r.In.Host)
+	// TODO: We're hard-coding "https" here,
+	// will need to be changed if "http" support is ever added.
+	r.Out.Header.Set("X-Forwarded-Proto", "https")
+	if c, ok := r.Out.Context().Value(serveHTTPContextKey{}).(*serveHTTPContext); ok {
+		r.Out.Header.Set("X-Forwarded-For", c.SrcAddr.Addr().String())
+	}
+}
+
+func (b *LocalBackend) addTailscaleIdentityHeaders(r *httputil.ProxyRequest) {
+	c, ok := r.Out.Context().Value(serveHTTPContextKey{}).(*serveHTTPContext)
+	if !ok {
+		return
+	}
+	node, user, ok := b.WhoIs(c.SrcAddr)
+	if !ok {
+		// Traffic originated from outside of Tailnet
+		// (funneled from public internet).
+		// No identity headers to add.
+		return
+	}
+	if !node.IsTagged() {
+		r.Out.Header.Set("X-Tailscale-User-Login", user.LoginName)
+		r.Out.Header.Set("X-Tailscale-User-Name", user.DisplayName)
+	} else {
+		r.Out.Header.Set("X-Tailscale-Tags", strings.Join(node.Tags, ","))
+	}
+	r.Out.Header.Set("X-Tailscale-Node", node.ComputedName)
```

Comment:
We should drop the "X-": https://www.rfc-editor.org/rfc/rfc6648

Nobody will conflict with "Tailscale-" prefixed headers. :)

Reply:
Updated to `X-Webauth-User` and `X-Webauth-Name` as more standard headers used by other services, such as grafana and gitea.

Reply:
Caught up with @bradfitz on Slack. I'm opting to go with `Tailscale-User-Login` and `Tailscale-User-Name` headers, and will add a followup commit making these easily configurable against other services, such as grafana. This should still keep things easily plug-and-play with common services, but lets us avoid being tied to other service's concepts of "users"/"logins". Also will allow us to keep consistent `Tailscale-` namings for tagged identity/other fields in the future that may not have matching equivalents in other services.

cc @willnorris who had some thoughts on header names.